rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java

```diff
@@ -451,31 +470,23 @@ private boolean getNotify(Block block, Keccak256 parentHash) {
     class NewBlockListener extends EthereumListenerAdapter {
 
         @Override
-        /**
-         * onBlock checks if we have to mine over a new block. (Only if the blockchain's best block changed).
-         * This method will be called on every block added to the blockchain, even if it doesn't go to the best chain.
-         * TODO(???): It would be cleaner to just send this when the blockchain's best block changes.
-         * **/
         // This event executes in the thread context of the caller.
         // In case of private miner, it's the "Private Mining timer" task
-        public void onBlock(Block block, List<TransactionReceipt> receipts) {
+        public void onBestBlock(Block newBestBlock, List<TransactionReceipt> receipts) {
```

Comment:
Could this change be done in a separate commit? Changing the hook is something I've been meaning to do for a while, but was never sure if could cause unintended consequences.
Having it as a separate commit (or even a separate PR) would be very helpful.

Reply:
I always meant to do this refactor separately but it ended up being a must for adding `MiningMainchainView` to mining flow. The reason for the former is that there was no easy way to add it using the `onBlock` hook. That explains why the change belongs to that commit.
As far as I'm aware, it's not causing any unintended consequences.

Reply:
Good to know